torch/quantization/fx/_equalize.py

```diff
@@ -395,20 +397,46 @@ def scale_weight_functional(
     if weight_observer_nodes is None:
         return
 
+    weight = None
     for weight_node in weight_observer_nodes:
         # Find the node containing the weight values
         if weight_node.op == 'get_attr':
             weight = model.get_buffer(str(weight_node.target))
+            break
+    if weight is None:
+        return
+
+    # Scale the weights for input-weight equalization
+    scaled_weight = torch.mul(weight, torch.reciprocal(equalization_scale))
+    weight_parent_name, weight_name = _parent_name(weight_node.target)
+
+    if next_equalization_scale is None:
+        setattr(modules[weight_parent_name], weight_name, scaled_weight)
+        assert(torch.allclose(model.get_buffer(str(weight_node.target)), scaled_weight))
+        return
+
+    # Multiply the weights row wise by the next equalization scale
+    new_shape = [1] * weight.ndim
+    new_shape[0] = weight.size(0)
+    scaled_weight = torch.mul(scaled_weight, next_equalization_scale.view(new_shape))
+
+    setattr(modules[weight_parent_name], weight_name, scaled_weight)
+    assert(torch.allclose(model.get_buffer(str(weight_node.target)), scaled_weight))
 
-            # Scale the weights for input-weight equalization
-            # If the following layer needs to be equalized then we will multiply its scale
-            scaled_weight = torch.mul(weight, torch.reciprocal(equalization_scale))
-            # TODO: connecting functional layers?
-            # scaled_weight = torch.mul(scaled_weight, next_equalization_scale)
+    # Multiply the bias element wise by the next equalization scale
+    bias = None
+    for bias_node, _ in op_node.users.items():
```

Comment:
how do we know this is bias?

Reply:
I was thinking that the nodes that go into the op_node are the bias, the weight equalization observer, and the input equalization observer. So the only on with the get_attr operation would be the bias. Maybe a better way to check is that the name contains the word "bias"?